File: test/parity.py
import json
import os
import sys

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
import pandas as pd
from indicators import ema, rsi, volume_ratio, last

here = os.path.dirname(os.path.abspath(__file__))
with open(os.path.join(here, "parity_data.json")) as f:
    d = json.load(f)

closes = pd.Series(d["closes"], dtype="float64")
volumes = pd.Series(d["volumes"], dtype="float64")

print(json.dumps({
    "ema9": last(ema(closes, 9)),
    "ema21": last(ema(closes, 21)),
    "rsi14": last(rsi(closes, 14)),
    "vol_ratio": volume_ratio(volumes, 20),
}, indent=2))

File: test/parity.mjs
import { readFileSync } from "node:fs";
import { fileURLToPath } from "node:url";
import { dirname, join } from "node:path";
import { emaLast, rsiLast, volumeRatio } from "../netlify/functions/scan.mjs";

const here = dirname(fileURLToPath(import.meta.url));
const d = JSON.parse(readFileSync(join(here, "parity_data.json"), "utf8"));

console.log(JSON.stringify({
  ema9: emaLast(d.closes, 9),
  ema21: emaLast(d.closes, 21),
  rsi14: rsiLast(d.closes, 14),
  vol_ratio: volumeRatio(d.volumes, 20),
}, null, 2));
